From 1a0ec5c30c37d29e4435a45e75c896f91af970bd Mon Sep 17 00:00:00 2001
From: JackieLiu <liuyun01@kylinos.cn>
Date: Tue, 29 Nov 2016 11:57:30 +0800
Subject: [PATCH] md/raid5-cache: do not need to set STRIPE_PREREAD_ACTIVE
 repeatedly
Git-commit: 1a0ec5c30c37d29e4435a45e75c896f91af970bd
Patch-mainline: v4.10-rc1
References: FATE#321488

R5c_make_stripe_write_out has set this flag, do not need to set again.

Signed-off-by: JackieLiu <liuyun01@kylinos.cn>
Signed-off-by: Shaohua Li <shli@fb.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/md/raid5-cache.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/md/raid5-cache.c b/drivers/md/raid5-cache.c
index a46f547..e786d4e 100644
--- a/drivers/md/raid5-cache.c
+++ b/drivers/md/raid5-cache.c
@@ -1247,8 +1247,6 @@ static void r5c_flush_stripe(struct r5conf *conf, struct stripe_head *sh)
 	atomic_inc(&conf->active_stripes);
 	r5c_make_stripe_write_out(sh);
 
-	if (!test_and_set_bit(STRIPE_PREREAD_ACTIVE, &sh->state))
-		atomic_inc(&conf->preread_active_stripes);
 	raid5_release_stripe(sh);
 }
 
-- 
2.10.2

